src/test/java/bio/terra/service/snapshot/SnapshotServiceTest.java

```diff
@@ -0,0 +1,179 @@
+        snapshotTableId = UUID.randomUUID();
+        profileId = UUID.randomUUID();
+        createdDate = Instant.now();
+    }
+
+    @Test
+    public void testRetrieveSnapshot() {
+        mockSnapshot();
+        assertThat(service.retrieveAvailableSnapshotModel(snapshotId), equalTo(
+            new SnapshotModel()
+                .id(snapshotId.toString())
+                .name(SNAPSHOT_NAME)
+                .description(SNAPSHOT_DESCRIPTION)
+                .createdDate(createdDate.toString())
+                .source(Collections.emptyList())
+                .tables(List.of(new TableModel()
+                    .name(SNAPSHOT_TABLE_NAME)
+                ))
+                .relationships(Collections.emptyList())
+                .profileId(profileId.toString())
+                .dataProject(SNAPSHOT_DATA_PROJECT)
+        ));
+    }
+
+    @Test
+    public void testRetrieveSnapshotNoFields() {
+        mockSnapshot();
+        assertThat(service.retrieveAvailableSnapshotModel(snapshotId, List.of(SnapshotRequestAccessInclude.NONE)),
+            equalTo(new SnapshotModel()
+                .id(snapshotId.toString())
+                .name(SNAPSHOT_NAME)
+                .description(SNAPSHOT_DESCRIPTION)
+                .createdDate(createdDate.toString())));
+    }
+
+    @Test
+    public void testRetrieveSnapshotDefaultFields() {
+        mockSnapshot();
+        assertThat(service.retrieveAvailableSnapshotModel(snapshotId, List.of(
+            SnapshotRequestAccessInclude.SOURCES,
+            SnapshotRequestAccessInclude.TABLES,
+            SnapshotRequestAccessInclude.RELATIONSHIPS,
+            SnapshotRequestAccessInclude.PROFILE,
+            SnapshotRequestAccessInclude.DATA_PROJECT)),
+            equalTo(service.retrieveAvailableSnapshotModel(snapshotId)));
+    }
+
+    @Test
+    public void testRetrieveSnapshotOnlyAccessInfo() {
+        mockSnapshot();
+        assertThat(service.retrieveAvailableSnapshotModel(snapshotId, List.of(
+            SnapshotRequestAccessInclude.ACCESS_INFORMATION)),
+            equalTo(new SnapshotModel()
+                .id(snapshotId.toString())
+                .name(SNAPSHOT_NAME)
+                .description(SNAPSHOT_DESCRIPTION)
+                .createdDate(createdDate.toString())
+                .accessInformation(new SnapshotAccessInfoModel()
+                    .bigQuery(new SnapshotAccessInfoModelBigQuery()
+                        .datasetName(SNAPSHOT_NAME)
+                        .projectId(SNAPSHOT_DATA_PROJECT)
+                        .link("https://console.cloud.google.com/bigquery?project=" + SNAPSHOT_DATA_PROJECT)
+                        .tables(List.of(new SnapshotAccessInfoModelBigQueryTable()
+                            .name(SNAPSHOT_TABLE_NAME)
+                            .sampleQuery("SELECT * FROM `" + SNAPSHOT_DATA_PROJECT + "." + SNAPSHOT_NAME + "." +
+                                SNAPSHOT_TABLE_NAME + "` LIMIT 1000")
```

Comment:
Not sure if you want to do this, but if the query string constant was protected instead of private, you could use it to build the match string here. That way if it ever changed (e.g. "LIMIT 2000") the test would still run.

Reply:
I actually like the idea of keeping the value separate to make sure that it's independently checking the string

Reply:
Sure, as long as it's testing something that's part of the contract of the code under test. If LIMIT 1000, for example, is a required feature that must be present, it should be tested for. If it's an implementation detail then putting it in the test creates an unnecessary link between the test and the code under test.